WordPress/gutenberg
# SandBox: Convert to TypeScript

Part of #35744

## What?

Add TypeScript types and documentation to the `SandBox` component.

## Why?

So we can ship TS data with the wp-components package.

## Testing Instructions

1. `npm run storybook:dev`
2. The Docs view for `SandBox` should look good, and the prop descriptions should match the README.md.
